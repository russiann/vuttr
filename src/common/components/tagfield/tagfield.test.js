import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';

import TagField from './tagfield.component';

afterEach(cleanup);

it('snapshot', () => {
  const {asFragment} = render(
    <TagField value={['lorem', 'ipsum', 'dolor', 'sit']} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should render tags', () => {
  const {asFragment, queryAllByTestId} = render(
    <TagField value={['lorem', 'ipsum', 'dolor', 'sit']} />
  );
  const tags = queryAllByTestId('tag');
  expect(asFragment()).toMatchSnapshot();
  expect(tags).toHaveLength(4);
});

it('should add tag when type Enter', () => {
  const onChangeSpy = jest.fn();
  const {getByTestId} = render(
    <TagField value={['js', 'css']} onChange={onChangeSpy} />
  );
  const textfield = getByTestId('textfield');
  fireEvent.change(textfield, {target: {value: 'node'}});
  fireEvent.keyDown(textfield, {keyCode: 13});
  expect(onChangeSpy).toHaveBeenCalledTimes(1);
  expect(onChangeSpy).toHaveBeenCalledWith(['js', 'css', 'node']);
});

it('should add tag when type comma', () => {
  const onChangeSpy = jest.fn();
  const {getByTestId} = render(
    <TagField value={['js', 'css']} onChange={onChangeSpy} />
  );
  const textfield = getByTestId('textfield');
  fireEvent.change(textfield, {target: {value: 'node'}});
  fireEvent.keyDown(textfield, {keyCode: 188});
  expect(onChangeSpy).toHaveBeenCalledTimes(1);
  expect(onChangeSpy).toHaveBeenCalledWith(['js', 'css', 'node']);
});

it('should remove tag when type backspace', () => {
  const onChangeSpy = jest.fn();
  const {getByTestId} = render(
    <TagField value={['js', 'css', 'node']} onChange={onChangeSpy} />
  );
  const textfield = getByTestId('textfield');
  fireEvent.keyDown(textfield, {keyCode: 8});
  expect(onChangeSpy).toHaveBeenCalledTimes(1);
  expect(onChangeSpy).toHaveBeenCalledWith(['js', 'css']);
});
