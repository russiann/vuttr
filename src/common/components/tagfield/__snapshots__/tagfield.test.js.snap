// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render tags 1`] = `
<DocumentFragment>
  .emotion-10 {
  font-family: 'Ubuntu',sans-serif;
  border: 1px dashed #ffb18f;
  border-radius: 4px;
  min-width: 200px;
  box-sizing: border-box;
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 10px;
  overflow: hidden;
}

.emotion-0 {
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background: #ffe5d6;
  padding: 0 2px;
  margin-right: 5px;
  color: #212121;
  cursor: pointer;
}

.emotion-8 {
  font-family: Ubuntu;
  outline: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  font-size: 15px;
}

<div
    class="emotion-10 emotion-11"
    type="text"
  >
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #lorem
    </span>
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #ipsum
    </span>
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #dolor
    </span>
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #sit
    </span>
    <input
      class="emotion-8 emotion-9"
      data-testid="textfield"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`snapshot 1`] = `
<DocumentFragment>
  .emotion-10 {
  font-family: 'Ubuntu',sans-serif;
  border: 1px dashed #ffb18f;
  border-radius: 4px;
  min-width: 200px;
  box-sizing: border-box;
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 10px;
  overflow: hidden;
}

.emotion-0 {
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background: #ffe5d6;
  padding: 0 2px;
  margin-right: 5px;
  color: #212121;
  cursor: pointer;
}

.emotion-8 {
  font-family: Ubuntu;
  outline: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  font-size: 15px;
}

<div
    class="emotion-10 emotion-11"
    type="text"
  >
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #lorem
    </span>
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #ipsum
    </span>
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #dolor
    </span>
    <span
      class="emotion-0 emotion-1"
      data-testid="tag"
    >
      #sit
    </span>
    <input
      class="emotion-8 emotion-9"
      data-testid="textfield"
      value=""
    />
  </div>
</DocumentFragment>
`;
